File: OlderCodes/Fortran77codes/random.f
c  From: "A SURVEY OF COMPUTATIONAL PHYSICS" 
c  by RH Landau, MJ Paez, and CC BORDEIANU 
c  Copyright Princeton University Press, Princeton, 2008.
c  Electronic Materials copyright: R Landau, Oregon State Univ, 2008;
c  MJ Paez, Univ Antioquia, 2008; and CC BORDEIANU, Univ Bucharest, 2008.
c  Support by National Science Foundation                                
c   
c	    random.f: simple random number generator, not for serious work
c  
      Program random
			
      	Implicit none
       	Integer i, number, old, seed, x, y
c                                  set seed, number generated numbers
       	seed = 11
       	number = 1000
       	Open(6, file = 'random.dat', Status = 'Unknown')
        old = seed
c                                                           execution
      	Do 10 i = 1, number
          x = Mod((57*old+1), 256)
          y = Mod((57*x+1), 256)
          Write (6, *) x, y
          old = y 
 10     Continue
        Close(6)
        Stop 'data saved in random.dat'
      End

File: OlderCodes/Ccodes/random.c
/* 
************************************************************************
*  random.c:    A simple random number generator, not for serious work *
*                                                                      *
*  From: "A SURVEY OF COMPUTATIONAL PHYSICS" 
   by RH Landau, MJ Paez, and CC BORDEIANU 
   Copyright Princeton University Press, Princeton, 2008.
   Electronic Materials copyright: R Landau, Oregon State Univ, 2008;
   MJ Paez, Univ Antioquia, 2008; & CC BORDEIANU, Univ Bucharest, 2008
   Support by National Science Foundation                              
*                                                                      *  
*  comment: plot without connecting datapoints with lines              *
************************************************************************
*/
#include <stdio.h>

#define max 1000                      /* number of numbers generated */
#define seed 11                       /* seed for number generator */

main ()
{
   int i, old, newx, newy;
   
   FILE *output;          /* save data in badrand.dat */
   output = fopen("badrand.dat","w");
   
   old = seed;                        /* the seed */

   for (i=0; i<max; i++)              /* generating #max numbers */
   {
      newx = (57*old+1) % 256;        /* x-coordinate */
      newy = (57*newx+1) % 256;       /* y-coordinate */
      fprintf (output, "%i\t%i\n", newx, newy);
      old  = newy;
   }
   printf("data stored in badrand.dat.\n");
   fclose(output);
}
 
